app/src/main/java/com/easyfitness/DAO/DAOMachine.java

```diff
@@ -203,6 +203,18 @@ public Cursor getAllMachines(int type) {
         return getMachineListCursor(selectQuery);
     }
 
+    /**
+     * @return List of Machine objects ordered by Favorite and Name given exercise types as input
+     */
+    public Cursor getAllMachines(String types) {
```

Comment:
As explained later, the parameters should rather be an array of ExerciseType.

Reply:
Explained above.